csrc/fusion.cpp

```diff
@@ -546,7 +547,9 @@ void Fusion::registerExpr(Expr* expr) {
   for (Val* input : expr->inputs()) {
     has_tv = has_tv || input->isA<TensorView>();
     assertInContainer(input, "Input to expr is invalid, ");
-    auto uses_copy = input->uses();
+    // Do not call uses() here, since we want to avoid resetTvUses unless
+    // necessary
+    auto uses_copy = input->uses_;
```

Comment:
This pattern of copying `uses_`, appending an `Expr` if not there and replacing `uses_` with the copy seems common. Why not adding a method to `Val` such as `addUse` that does this? We also can skip copying the use vector.

Reply:
Thanks @naoyam . Avoiding those copies seems to have given us a little bit extra boost: 
![image](https://user-images.githubusercontent.com/1454944/229898497-8390b655-87fd-48d4-8517-b0c370ca4c69.png)